src/modules/ContentCard.js

```diff
@@ -0,0 +1,188 @@
+// (C) Copyright 2014-2015 Hewlett Packard Enterprise Development LP
+
+import React, { Component, PropTypes } from 'react';
+import classnames from 'classnames';
+import Props from 'grommet/utils/Props';
+import Box from 'grommet/components/Box';
+import Tile from 'grommet/components/Tile';
+import Heading from 'grommet/components/Heading';
+import Paragraph from 'grommet/components/Paragraph';
+import Anchor from 'grommet/components/Anchor';
+import Layer from 'grommet/components/Layer';
+import Video from 'grommet/components/Video';
+import LinkNextIcon from 'grommet/components/icons/base/LinkNext';
+import PlayIcon from 'grommet/components/icons/base/Play';
+
+const CLASS_ROOT = 'content-card';
+
+export default class ContentCard extends Component {
+  constructor (props) {
+    super(props);
+    this._handleClick = this._handleClick.bind(this);
+    this.state = {
+      activeVideo: false
+    };
+  }
+
+  _handleClick (event) {
+    const { video, link } = this.props;
+
+    if (video) {
+      event.preventDefault();
+      this.setState({ activeVideo : !this.state.activeVideo });
+    } else if (link) {
+      window.location.href = link;
+    }
+  }
+
+  _renderChildren () {
+    const { children, socialIcon } = this.props;
+
+    if (socialIcon) {
+      return (
+        <Box className="flex" pad={{between: 'large'}}>
+          {children}
+          <Box align="end">
+            {socialIcon}
+          </Box>
+        </Box>
+      );
+    }
+
+    return children;
+  }
+
+  _renderLinkMarkup () {
+    const { link, linkIcon, linkText, onClick, video } = this.props;
+
+    let linkMarkup;
+    let anchorLabel = linkText;
+    if (!linkText) {
+      anchorLabel = video ? 'Watch Now' : 'Learn More';
+    }
+
+    let anchorIcon = linkIcon;
+    if (!linkIcon) {
+      anchorIcon = video ? <PlayIcon /> : <LinkNextIcon />;
+    }
+
+    if (link || onClick || video) {
+      linkMarkup = (
+        <Box pad={{vertical: "small"}}>
+          <Anchor href={link} icon={anchorIcon} label={anchorLabel}
+            onClick={this._handleClick} />
+        </Box>
+      );
+    }
+
+    return linkMarkup;
+  }
+
+  _renderVideoMarkup () {
+    const { video } = this.props;
+    const { activeVideo } = this.state;
+    let videoMarkup;
+
+    if (video && video.source && activeVideo) {
+      videoMarkup = (
+        <Layer onClose={this._handleClick} closer={true} flush={true}>
+          <Video>
+            <source src={video.source} type={`video/${video.type}`}/>
+          </Video>
+        </Layer>
+      );
+    }
+
+    return videoMarkup;
+  }
+
+  render () {
+    const { thumbnail, description, heading, overline, link, onClick,
+      video, socialIcon, direction, contentPlacement, pad } = this.props;
+    const tileProps = Props.pick(this.props, Object.keys(Tile.propTypes));
+    delete tileProps.onClick;
```

Comment:
If you follow the [comment above](https://github.com/grommet/hpe-digitaltoolkit/pull/85#discussion_r73319626), we wont need to `delete tileProps.onClick`.